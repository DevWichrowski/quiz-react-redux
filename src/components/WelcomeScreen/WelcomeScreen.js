import React from 'react';
import "./WelcomeScreen.scss";
import Header from "../shared/Header/Header";
import Button from "../shared/Button/Button";
import {useHistory} from 'react-router-dom';
import {connect} from "react-redux";
import {startQuiz} from "../../store/actions/quiz.actions";

const WelcomeScreen = ({start}) => {
    const history = useHistory();

    const startQuiz = () => {
        history.push("/quiz");
        start()
    };

    return (
        <div className="welcome-screen">
            <Header headerText={"Welcome in React Quiz"}/>
            <div className="welcome-screen__content">
                <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquam, aliquid animi cum deleniti
                    doloremque ea eveniet illum ipsum iure maiores minus modi molestiae necessitatibus optio placeat
                    quaerat quis quos sapiente?</p>
            </div>

            <Button buttonText={"Start Quiz"} onClick={startQuiz}/>
        </div>
    );
};

const mapDispatchToProps = dispatch => ({
    start: () => dispatch(startQuiz())
});

export default connect(null, mapDispatchToProps)(WelcomeScreen)