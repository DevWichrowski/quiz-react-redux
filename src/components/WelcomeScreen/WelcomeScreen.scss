.welcome-screen {
  display: block;
  height: 100%;

  ul {
    //margin-top: 30px;
    padding: 5%;
    margin: 0 auto;

    li {
      margin-bottom: 10px;
    }
  }
}