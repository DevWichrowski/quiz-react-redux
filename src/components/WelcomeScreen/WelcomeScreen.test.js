import WelcomeScreen from "./WelcomeScreen";
import {testConnectedComponentRender} from "../../core/helpers/tests/test-connected-component-render";


describe('WelcomeScreen', () => {
    testConnectedComponentRender(WelcomeScreen)
});
