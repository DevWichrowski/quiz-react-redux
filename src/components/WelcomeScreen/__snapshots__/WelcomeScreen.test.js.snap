// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen should render 1`] = `
<WelcomeScreen
  start={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
/>
`;
